### DISCUSSION TITLE
Trouble with cross sections and kwargs

### DISCUSSION THREAD

**Discussion by hinklePSI** [gdsfactory/gdsfactory#2400] [Q&A] [answered]
Hi all,
I am having trouble adjusting certain parameters such as the width of the waveguides on mzm's and other larger components. For instance, in the code below, I am trying to change the bend parameter:

```
bend = gf.components.bend_s(size=[10,5],  npoints=99, cross_section='xs_sc', with_bbox=True, add_pins=True)              
mzi = gf.components.mzi(length_x = 23635, length_y = 0, splitter = m2, delta_length = 0, bend=bend)
```

I am seeing the following error:
    raise ValueError(f"Cannot apply kwargs {kwargs} to {component.name!r}")
ValueError: Cannot apply kwargs {'cross_section': 'xs_sc'} to 'bend_s_size10__5'

Any idea why this is happening? 

Thanks.

**Comment by joamatab** [comment, marked as answer]
Hi Max

thank you for the message, yes, the message was not very clear and we were being a bit too strict

in this PR i convert the error into a warning with a better message
```
      f"Cannot apply settings {kwargs} to already created component {component.name!r}."
      "Try passing a function instead of a Component."
```

this is the recommend way of doing this
```
import gdsfactory as gf 
from functools import partial 

bend90 = partial(gf.components.bend_euler, radius=10, angle=90, npoints=99, cross_section='xs_sc') # this is still a function
mzi = gf.components.mzi(length_x = 236, length_y = 10, delta_length = 0, bend=bend90) # mzi expects a function for the bend
mzi.show()
```

here is the PR with the improvements
https://github.com/gdsfactory/gdsfactory/discussions/2400

**Reply by hinklePSI** [reply]
Thank you very much for this, everything is working now.
